src/parseTools.js

```diff
@@ -37,10 +37,12 @@ function processMacros(text) {
 function preprocess(text, filenameHint) {
   if (EXPORT_ES6 && USE_ES6_IMPORT_META) {
     // `eval`, Terser and Closure don't support module syntax; to allow it,
```

Comment:
Yes, eventually it would be nice to fix Terser and Closure, but it's a relatively large change.

For Terser we need to cherry-pick https://github.com/terser/terser/commit/24ce297dfc6b298251d031bc9a4f89c005a853fa on top of https://github.com/emscripten-core/terser/tree/emscripten_patches_v4.8.0 and Closure needs to run with some additional flags.
https://github.com/emscripten-core/emscripten/blob/234940f032e4353a4390af02bbc59a6a9f87afcf/tools/building.py#L603-L610

Though, I think(?), `eval()` still needs this replacement, since `import.meta` is used in `library_pthread.js` as well.
https://github.com/emscripten-core/emscripten/blob/234940f032e4353a4390af02bbc59a6a9f87afcf/src/modules.js#L215-L222
https://github.com/emscripten-core/emscripten/blob/234940f032e4353a4390af02bbc59a6a9f87afcf/emscripten.py#L207-L210

The above changes are not included in this PR, see comment https://github.com/emscripten-core/emscripten/pull/17915#issuecomment-1311871913.

Reply:
> Though, I think(?), `eval()` still needs this replacement, since `import.meta` is used in `library_pthread.js` as well.

Pretty sure this was the reason I used this hack in the first place btw - it was just easier to support the variety of tooling + `eval` this way instead of doing something custom for each of them separately.

Reply:
> Pretty sure this was the reason I used this hack in the first place btw - it was just easier to support the variety of tooling + `eval` this way instead of doing something custom for each of them separately.

👍, I realised that too late, after doing all those Terser and Closure changes. 😅

Anyway, the Closure change, that optimizes the wrapper function when linking with `-sMODULARIZE`, might still be useful. I'll see if I can split that up after this PR lands.